flang/lib/Lower/Bridge.cpp

```diff
@@ -340,21 +340,22 @@ class FirConverter : public Fortran::lower::AbstractConverter {
         llvm::None);
   }
 
-  bool createHostAssociateVarClone(
-      const Fortran::semantics::Symbol &sym) override final {
+  bool createHostAssociateVarClone(const Fortran::semantics::Symbol &sym,
+                                   bool init) override final {
```

Comment:
I'm a little unclear on why the variable/symbol is being initialized in the code that adds it to the symbol table(map). Is it possible to keep these two steps separate?

Reply:
OK. I have split the function into two.